src/components/ValidatedInput/ValidFieldPassword.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ *  * Copyright (c) 2021 AccelByte Inc. All Rights Reserved.
+ *  * This is licensed software from AccelByte Inc, for limitations
+ *  * and restrictions contact your company contract manager.
+ *
+ */
+
+import * as React from "react";
+import { ValidFieldText, ValidFieldTextProps } from "./ValidFieldText";
+import classNames from "classnames";
+import ReactTooltip from "react-tooltip";
+import "./ValidFieldPassword.scss";
+import { generatePassword } from "../../utils/utils";
+import { DEFAULT_PASSWORD_AND_SECRET_REGEX } from "../../constants/common";
+
+export interface ValidFieldPasswordProps extends ValidFieldTextProps {
+  strengthLevelIndicator?: string;
+  passHideText: string;
+  passVisibleText: string;
+  hasGeneratePassword?: boolean;
+  defaultGenerateText?: string;
+  customPattern?: string;
+}
+
+interface State {
+  isIconEyeOff: boolean;
+  levelColor: string;
+}
+
+export class ValidFieldPassword extends React.Component<ValidFieldPasswordProps, State> {
+  constructor(props: ValidFieldPasswordProps) {
+    super(props);
+    this.state = {
+      isIconEyeOff: true,
+      levelColor: "",
+    };
+
+    if (props.type === "number") {
+      throw new Error('ValidFieldPassword Component cannot be used when the prop type="number"');
+    }
+  }
+
+  toolTipIconEye = React.createRef<HTMLElement>();
+
+  componentDidMount() {
+    if (this.props.strengthLevelIndicator) this.handleStrengthIndicator();
+    setTimeout(() => {
+      ReactTooltip.rebuild();
+    }, 100);
+  }
+
+  componentWillUnmount() {
+    this.hideTooltip();
+  }
+
+  handleStrengthIndicator = () => {
+    const { strengthLevelIndicator } = this.props;
+    switch (strengthLevelIndicator) {
+      case "Poor!":
```

Comment:
is it possible to use enum for each `strengthLevelIndicator` and `levelColor`?

Reply:
It should be able to do so, let me try it first

Reply:
updated `Enum` [here](https://github.com/AccelByte/justice-ui-library/pull/1/files#diff-cf49d988109f199a1886a4bca8510934e281c8334241c762d07ef55d6b311624R11) and the component usage [here](https://github.com/AccelByte/justice-ui-library/pull/1/files#diff-c1422bbfca4bb5ad5899c943e234574286157befa0dfc91caab16500f20a5a97R57)